src/OrchardCore.Modules/OrchardCore.AuditTrail/Providers/IAuditTrailEventProvider.cs

```diff
@@ -0,0 +1,15 @@
+﻿using OrchardCore.AuditTrail.Services.Models;
+
+namespace OrchardCore.AuditTrail.Providers
+{
+    /// <summary>
+    /// Audit Trail Event Provider implementations can describe events contributing to Audit Trail.
+    /// </summary>
+    public interface IAuditTrailEventProvider
```

Comment:
If this is meant to be reused by other modules it should be an abstractions project.